app/grandchallenge/documentation/models.py

```diff
@@ -0,0 +1,149 @@
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import models
+from django.db.models import Max
+from django.utils.translation import gettext_lazy as _
+from django_extensions.db.fields import AutoSlugField
+from simple_history.models import HistoricalRecords
+
+from grandchallenge.core.templatetags.bleach import clean
+from grandchallenge.core.utils.query import index
+from grandchallenge.subdomains.utils import reverse
+
+
+class LevelChoices(models.IntegerChoices):
+    """Documentation page level options"""
+
+    PRIMARY = 1, _("Top level")
+    SECONDARY = 2, _("Second level")
+    TERTIARY = 3, _("Third level")
+
+
+class DocPageLevel:
+    """Documentation page level."""
+
+    DocPageLevelChoices = LevelChoices
+
+
+class DocPage(models.Model):
+
+    Level = DocPageLevel.DocPageLevelChoices
+
+    UP = "UP"
+    DOWN = "DOWN"
+    FIRST = "FIRST"
+    LAST = "LAST"
+
+    created = models.DateTimeField(auto_now_add=True)
+    modified = models.DateTimeField(auto_now=True)
+
+    title = models.CharField(max_length=1024)
+    slug = AutoSlugField(populate_from="title", max_length=1024)
+    display_title = models.CharField(
+        max_length=255,
+        default="",
+        blank=True,
+        help_text=(
+            "On pages and in menu items, use this text. Spaces and special "
+            "chars allowed here. Optional field. If emtpy, title is used"
+        ),
+    )
+
+    content = models.TextField()
+
+    level = models.PositiveSmallIntegerField(

```

Comment:
If we only had top and second level pages, we wouldn't need the levels anymore, but if we also want to have a third level, I think the levels field will be necessary to tell second from third level pages. Both second and third level pages have a parent and while only second level pages can have children, they don't necessarily need to have children, so I think it would be difficult to retrieve third level pages seperately unless we have an explicit way of marking them as third level pages.

Reply:
It's hard to say without seeing how this would be used in practice. There are approaches that would work without having to maintain the levels/parents but it depends on how the model is going to be used. What problem are you trying to solve here? It is really hard to judge what approach we should take just from these tests. If it is for a sidebar then we need to think of a data structure that can deliver all of the information we need in a single call, so I wonder if getting all the pages by level is necessary.

Reply:
Hm, yes, this is only for the sidebar, so for creating a visual hierarchy of pages. I've now pushed what I had in mind for the detail view. I'm not sure how we would combine the level and the parent/children information into just one field on the model.

Reply:
I think I found a way to not use the levels field after all. I should have seen this earlier...